SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/Tree/Node.swift

```diff
@@ -47,7 +47,31 @@ class Node {
 		for childNode in self.childNodes {
 			childNode.unHighlightSubTree()
 		}
-	}	
+	}
+    
+    // it will retrun a node. if it has only nonDummy child node, it will return that child node, else it will return itself
+    func mergeNode() -> Node {
+        var nonDummyChildCount = 0
+        var nonDummyChildren : Node = Node()
+        for childNode in self.childNodes {
+            if childNode.dummy == false {
```

Comment:
This function can be made simpler if you change this if statement to
```swift
if childNode.dummy == false {
    if nonDummyChild == nil {
        nonDummyChild = childNode
    } else {
        return self;
    }
}
```
Then right after the for loop, just
`return nonDummyChild!`
Originally (line 55), don't set `nonDummyChild` to `Node()`, rather set it to nil.
Now you don't need the `unDummyNum` variable (which probably should've been called something like `nonDummyChildCount`), and you have fewer lines overall! :)

"nil requires a contextual type" 

Right. So you need to specify the type.
`var nonDummyChild:Node? = nil`
and eventually `return nonDummyChild!`